zipkin/src/main/java/zipkin/filter/SpanFilter.java

```diff
@@ -0,0 +1,24 @@
+package zipkin.filter;
+
+import zipkin.Span;
+import zipkin.storage.Callback;
+
+import java.util.List;
+
+public interface SpanFilter {
+  /**
+   * Process filters given a set of v1 spans. The callback should return a FilterActivatedException if filter
+   * implementor wants to produce custom return codes back to the user.
+   * @param spans
+   * @param callback
+   */
```

Comment:
hmm are you hinting there is a partial success here? ex you can return without exception?

Reply:
If the implementor so chooses they could consider ingesting all spans, but throwing an error. I think that would be a rare case, but I can't think of reasons to get in someone's way of doing that; nor can I think of a strong reason to go out of our way to support that very well. Do you want me to have the call of the interface stop if exception is added to callback?